src/app.rs

```diff
@@ -713,57 +713,65 @@ impl SsbhApp {
         let final_frame_index = self.max_final_frame_index();
 
         // TODO: Find a better layout for this.
-        ui.checkbox(&mut self.animation_state.should_loop, "Loop");
-        ui.label("Speed");
-        ui.add(
-            DragValue::new(&mut self.animation_state.playback_speed)
-                .min_decimals(2)
-                .speed(0.01)
-                .clamp_range(0.25..=2.0),
-        );
+        ui.vertical(|ui| {
+            ui.horizontal(|ui| {
+                ui.label("Speed");
+                ui.add(
+                    DragValue::new(&mut self.animation_state.playback_speed)
+                        .min_decimals(2)
+                        .speed(0.01)
+                        .clamp_range(0.25..=2.0),
+                );
 
-        // TODO: How to fill available space?
-        // TODO: Get the space that would normally be taken up by the central panel?
-        ui.spacing_mut().slider_width = (ui.available_width() - 520.0).max(0.0);
-        if ui
-            .add(
-                // TODO: Show ticks?
-                egui::Slider::new(
-                    &mut self.animation_state.current_frame,
-                    0.0..=final_frame_index,
-                )
-                .step_by(1.0)
-                .show_value(false),
-            )
-            .changed()
-        {
-            // Manually trigger an update in case the playback is paused.
-            self.animation_state.should_update_animations = true;
-        }
+                // TODO: Custom checkbox widget so label is on the left side.
+                ui.checkbox(&mut self.animation_state.should_loop, "Loop");
+            });
+            ui.horizontal(|ui| {
```

Comment:
I would use horizontal_centered here to center the slider and buttons. Centering both layouts seems to cause problems for some reason.